### DISCUSSION TITLE
Индивидуальные голосовые приветствия голосовой почты для внутренних абонентов

### DISCUSSION THREAD

**Discussion by DenisGaidamak** [mikopbx/Core#425] [Ideas (Идеи)]
Как насчёт модернизации существующего алгоритма работы голосовой почты с целью создания возможности погрузки индивидуальных голосовых приветствий абонентов? Остальной алгоритм оставить без изменений (оставленное абоненту сообщение отправляется на его e-mail и удаляется с сервера).

**Comment by boffart** [comment]
Мне вот реально интересен кейс, когда клиент реально будет мотивирован оставить голосовое сообщение )))
По мне, так большинство бросит трубку. 
Можете описать примеры процессов в вашей организации?

**Comment by boffart** [comment]
Технически, возможно организовать произвольное количество ящиков голосовой почты. 
Для этого нужно поправить файл `/etc/asterisk/voicemail.conf `  можно через "Кастомизация системных файлов". 

```
[other-104]
directoryintro=dir-company2
104 => 104,пользователь,test@gmail.com,,attach=yes|tz=local|delete=yes
```
В каталоге "**dir-company2**" должен лежать файл `vm-intro.gsm` который будет воспроизводится клиенту.  Расширение может быть и **wav**. 

И направлять вхоящий на этот голосовой ящик через "Приложение dialplan"
`1,Goto(voice-mail-other-104,voicemail,1)`

Пример контекста "`voice-mail-other-104`":

```
[voice-mail-other-104] 
exten => voicemail,1,Answer()
	same => n,ExecIf($["${CHANNEL:0:5}" == "Local"]?Set(pl=${IF($["${CHANNEL:-1}" == "1"]?2:1)}))
	same => n,ExecIf($["${CHANNEL:0:5}" == "Local"]?Set(bridgePeer=${IMPORT(${CUT(CHANNEL,\;,1)}\;${pl},BRIDGEPEER)}))
	same => n,ExecIf($[ "${FROM_CHAN}" == "${bridgePeer}" ]?ChannelRedirect(${bridgePeer},${CONTEXT},${EXTEN},2))
	same => n,AGI(/usr/www/src/Core/Asterisk/agi-bin/clean_timeout.php)
	same => n,Gosub(voicemail_start,${EXTEN},1)
	same => n,VoiceMail(104@other-104)
	same => n,Hangup()
```

**Comment by DenisGaidamak** [comment]
Ну у нас, например, врачи сами записывают пациентов на приём. И запрос на индивидуальное приветствие есть. Ну и вообще, как-то оно правильнее мне кажется...

Скажите пожалуйста, в каком каталоге нужно создать директорию "dir-company2" с vm-intro? Файловая система не совсем привычна...

**Comment by boffart** [comment]
Свои служебные файлу следует размещать в каталоге:
` /storage/usbdisk1/mikopbx/custom_modules`
особенность этого каталога - он всегда будет попадать в бекап при использовании модуля "Резервное копирование"

**Comment by DenisGaidamak** [comment]
Алексей, что-то мне подсказывает, что опция directoryintro - это приглашение телефонного справочника, а не индивидуального приветствия для контекста. Так как астериск никак не реагирует на эту опцию, а воспроизводит дефолтное основное приветствие.

Да, так и есть по ходу:

; For the directory, you can override the intro file if you want 
;directoryintro=dir-intro

Речь по ходу идёт о замене приветствия при использовании голосового телефонного справочника.

Думаю, там нужно раскладывать приветствия в папки ящиков голосовой почты. Что-то типа unavail.wav и busy.wav. Обычно они лежат в /var/spool/asterisk/voicemail/default/внутренний_номер/
